src/main.py

```diff
@@ -235,13 +235,19 @@ def parse_transaction(b64_tx, success_callback=None, error_callback=None):
         cancel=cb_with_args(error_callback, "user cancel")
     )
 
+def qr_animated_received():
+    scan_transaction()
+
+def qr_animated(indx):
+    gui.alert("Animated QR: %s/%s" % (indx[0], indx[1]), "Proceed to scanning code No. %s" % str(indx[0]+1), qr_animated_received)
+
 @catchit
 def scan_transaction():
     screens.show_progress("Scan transaction to sign",
                           "Scanning.. Click \"Cancel\" to stop.",
                           callback=cancel_scan)
     gui.update(30)
-    qr_scanner.start_scan(parse_transaction)
+    qr_scanner.start_scan(parse_transaction, qr_animated)
```

Comment:
We can avoid extra callbacks in global scope if we do the following:
```py
def qr_animated(indx, callback):
    gui.alert("Animated QR: %s/%s" % (indx[0], indx[1]), 
              "Proceed to scanning code No. %s" % str(indx[0]+1), 
              callback)

@catchit
def scan_transaction():
    screens.show_progress("Scan transaction to sign",
                          "Scanning.. Click \"Cancel\" to stop.",
                          callback=cancel_scan)
    gui.update(30)
    qr_scanner.start_scan(parse_transaction, qr_animated, scan_transaction)
```

Makes sense to add the same to add new wallet function:
```py
@catchit
def add_new_wallet():    
    screens.show_progress("Scan wallet to add",
                          "Scanning.. Click \"Cancel\" to stop.",
                          callback=cancel_scan)
    gui.update(30)
    qr_scanner.start_scan(parse_new_wallet, qr_animated, add_new_wallet)
```